src/main/java/com/epam/processor/DataProcessor.java

```diff
@@ -1,123 +1,240 @@
-        return null;
-    }
-
-
-    /**
-     * filter list by longtitude and latitude fields
-     * @param minLongitude
-     * @param maxLongitude
-     * @param minLatitude
-     * @param maxLatitude
-     * @return
-     */
-    public Collection<RoadAccident> getAccidentsByLocation(float minLongitude, float maxLongitude, float minLatitude, float maxLatitude){
-        return null;
-    }
-
-    /**
-     * find the weather conditions which caused max number of incidents
-     * @return
-     */
-    public List<String> getTopThreeWeatherCondition(){
-        return null;
-    }
-
-    /**
-     * count incidents by road surface conditions
-     * @return
-     */
-    public Map<String, Long> getCountByRoadSurfaceCondition(){
-        return null;
-    }
-
-    /**
-     * To match streaming operations result, return type is a java collection instead of multimap
-     * @return
-     */
-    public Map<String, List<String>> getAccidentIdsGroupedByAuthority(){
-        return null;
-    }
-
-}
+package com.epam.processor;
+
+import com.epam.data.AccidentsDataLoader;
+import com.epam.data.RoadAccident;
+import com.epam.data.RoadAccidentBuilder;
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.Multimap;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * This is to be completed by mentees
+ */
+public class DataProcessor {
+	private static final String ACCIDENTS_CSV = "src/main/resources/DfTRoadSafety_Accidents_2009.csv";
+
+	private final List<RoadAccident> roadAccidentList;
+
+	public DataProcessor(List<RoadAccident> roadAccidentList) {
+		this.roadAccidentList = roadAccidentList;
+	}
+
+	// First try to solve task using java 7 style for processing collections
+
+	/**
+	 * Return road accident with matching index
+	 * 
+	 * @param index
+	 * @return
+	 */
+	public RoadAccident getAccidentByIndex7(String index) {
+		for (RoadAccident roadAccident : roadAccidentList) {
+			if (roadAccident.getAccidentId().equals(index)) {
+				return roadAccident;
+			}
+		}
+		return null;
+	}
+
+	/**
+	 * filter list by longtitude and latitude values, including boundaries
+	 * 
+	 * @param minLongitude
+	 * @param maxLongitude
+	 * @param minLatitude
+	 * @param maxLatitude
+	 * @return
+	 */
+	public Collection<RoadAccident> getAccidentsByLocation7(float minLongitude,
+			float maxLongitude, float minLatitude, float maxLatitude) {
+		List<RoadAccident> list = new ArrayList<>();
+		for (RoadAccident roadAccident : roadAccidentList) {
+			if (roadAccident.getLongitude() >= minLongitude
+					&& roadAccident.getLongitude() <= maxLongitude
+					&& roadAccident.getLatitude() >= minLatitude
+					&& roadAccident.getLatitude() <= maxLatitude) {
+				list.add(roadAccident);
+			}
+		}
+		return list;
+	}

```

Comment:
@zero1608 
This is good. Try to use some meaningful names for variables. On place of list you could use accidentList.